mmgen/models/ddpm/denoising.py

```diff
@@ -0,0 +1,400 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from copy import deepcopy
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import constant_init
+from mmcv.cnn.bricks.conv_module import ConvModule
+from mmcv.runner import load_checkpoint
+
+from mmgen.models.builder import MODULES, build_module
+from mmgen.utils import get_root_logger
+from .modules import EmbedSequential
+
+
+@MODULES.register_module()
+class DenoisingUnet(nn.Module):
+    """Denoising Unet. This network receives a diffused image ``x_t`` and
+    current timestep ``t``, and returns a ``output_dict`` corresponding to the
+    passed ``output_cfg``.
+
+    ``output_cfg`` define the number of channels and the meaning of the output.
+    ``output_cfg`` support ``mean`` and ``var`` for keys, and
+    denotes how the the network output mean and variance required for denoising
+    process.
```

Comment:
``output_cfg`` mainly contains keys of ``mean`` and ``var``, denoting how the network outputs mean and variance required for the denoising process.